scripts/git/google_to_main.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Checks out a branch at the specified green commit (default "google") and
+# creates a PR from that branch based on main.
+#
+# - Requries the gh CLI (https://github.com/cli/cli) to create a PR.
+# - Will force push to the configured PR_BRANCH (default "google-to-main") on
+#   the configured FORK_REMOTE (default "fork")
+# - Requires that local "google" branch is a pristine (potentially stale) copy
+#   of the "google" branch on the configured UPSTREAM_REMOTE
+#   (default "upstream").
+# - Requires that the working directory be clean. Will abort otherwise.
+
+set -x
+set -e
+set -o pipefail
+
+GREEN_COMMIT="${1:-google}"
+
+PR_BRANCH="${PR_BRANCH:-google-to-main}"
+UPSTREAM_REMOTE="${UPSTREAM_REMOTE:-upstream}"
+FORK_REMOTE="${FORK_REMOTE:-fork}"
+
+if [[ -n "$(git status --porcelain)" ]]; then
+  echo "Working directory not clean. Aborting"
+  git status
+  exit 1
+fi
+if ! git symbolic-ref -q HEAD; then
+  echo "In a detached HEAD state. Aborting"
+  git status
+  exit 1
+fi
+git checkout google
+git pull "${UPSTREAM_REMOTE?}" google --ff-only
+git submodule update --init
+if [[ -n "$(git status --porcelain)" ]]; then
+  echo "Working directory not clean after sync. Aborting"
+  git status
+  exit 1
+fi
+if [[ "${GREEN_COMMIT}" != "google" ]]; then
+  git checkout "${GREEN_COMMIT?}"
+  git submodule update
+fi
+git checkout -B "${PR_BRANCH?}"
```

Comment:
Depending on how the diff appears on GitHub and if there are conflicts, a `git merge main` here could help. That may requite manual intervention and can happen after creating the PR (just adding the new commit(s) on top of the new branch). This script may not need to handle that case, but we can make a note in our docs.

Reply:
Yeah I think I asked about that earlier, whether we want to just unconditionally merge from the branch being merged into. The only disadvantage is noisier history...

Reply:
We could check for merge conflicts with something like 

```shell
if git merge --no-commit main; then
  git merge --abort
fi
```

But I don't know how we'd continue the script after.

Reply:
Let's leave it as-is for now and then explore always merging as we continue to play with this.